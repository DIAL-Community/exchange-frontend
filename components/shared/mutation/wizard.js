import { gql } from '@apollo/client'

export const CREATE_WIZARD_GUIDANCE_MAIL = gql`
  mutation CreateWizardGuidanceMail (
    $name: String!
    $emailAddress: String!
    $message: String!
  ) {
    createWizardGuidanceMail(
      name: $name
      emailAddress: $emailAddress
      message: $message
    ) {
      response
    }
  }
`
