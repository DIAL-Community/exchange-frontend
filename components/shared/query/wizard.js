import { gql } from '@apollo/client'

export const WIZARD_BUILDING_BLOCKS_QUERY = gql`
  query WizardBuildingBlocks(
    $sdgs: [String!]
    $useCases: [String!]
    $filterBlocks: [String!]
    $limit: Int!
    $offset: Int!
  ) {
    paginatedBuildingBlocks(
      sdgs: $sdgs
      useCases: $useCases
      filterBlocks: $filterBlocks
      offsetAttributes: { limit: $limit, offset: $offset }
    ) {
      id
      name
      slug
      imageFile
      maturity
    }
    paginationAttributeBuildingBlock(
      sdgs: $sdgs
      useCases: $useCases
      filterBlocks: $filterBlocks
    ) {
      totalCount
    }
  }
`

export const WIZARD_USE_CASES_QUERY = gql`
  query WizardUseCases(
    $useCases: [String!]
    $sectors: [String!]
    $buildingBlocks: [String!]
    $sdgs: [String!]
    $limit: Int!
    $offset: Int!
  ) {
    paginatedWizardUseCases(
      sdgs: $sdgs
      sectors: $sectors
      useCases: $useCases
      buildingBlocks: $buildingBlocks
      offsetAttributes: { limit: $limit, offset: $offset }
    ) {
      id
      slug
      name
      imageFile
      maturity
    }
    paginationWizardAttributeUseCase(
      sdgs: $sdgs
      useCases: $useCases
      sectors: $sectors
      buildingBlocks: $buildingBlocks
    ) {
      totalCount
    }
  }
`

export const WIZARD_PRODUCTS_QUERY = gql`
  query WizardProducts(
    $useCases: [String!]
    $buildingBlocks: [String!]
    $sectors: [String!]
    $tags: [String!]
    $sdgs: [String!]
    $limit: Int!
    $offset: Int!
  ) {
    paginatedProducts(
      useCases: $useCases
      buildingBlocks: $buildingBlocks
      sectors: $sectors
      tags: $tags
      sdgs: $sdgs
      offsetAttributes: { limit: $limit, offset: $offset }
    ) {
      id
      name
      slug
      tags
      imageFile
    }
    paginationAttributeProduct(
      useCases: $useCases
      buildingBlocks: $buildingBlocks
      sectors: $sectors
      tags: $tags
      sdgs: $sdgs
    ) {
      totalCount
    }
  }
`

export const WIZARD_PROJECTS_QUERY = gql`
  query WizardProjects(
    $countries: [String!]
    $sectors: [String!]
    $tags: [String!]
    $sdgs: [String!]
    $limit: Int!
    $offset: Int!
  ) {
    paginatedProjects(
      countries: $countries
      sectors: $sectors
      tags: $tags
      sdgs: $sdgs
      offsetAttributes: { limit: $limit, offset: $offset }
    ) {
      id
      name
      slug
      organizations {
        id
        slug
        name
        imageFile
      }
      products {
        id
        slug
        name
        imageFile
      }
    }
    paginationAttributeProject(
      countries: $countries
      sectors: $sectors
      tags: $tags
      sdgs: $sdgs
    ) {
      totalCount
    }
  }
`

export const WIZARD_DATASETS_QUERY = gql`
  query WizardDatasets(
    $sdgs: [String!]
    $tags: [String!]
    $sectors: [String!]
    $limit: Int!
    $offset: Int!
  ) {
    paginatedDatasets(
      sdgs: $sdgs
      tags: $tags
      sectors: $sectors
      offsetAttributes: { limit: $limit, offset: $offset }
    ) {
      id
      name
      slug
      imageFile
    }
    paginationAttributeDataset(
      sdgs: $sdgs
      tags: $tags
      sectors: $sectors
    ) {
      totalCount
    }
  }
`

export const WIZARD_EXTENDED_DATA_QUERY = gql`
  query WizardExtendedData {
    wizard {
      digitalPrinciples {
        phase
        name
        slug
        url
      }
      resources {
        phase
        name
        imageUrl
        link
        description
      }
    }
  }
`
