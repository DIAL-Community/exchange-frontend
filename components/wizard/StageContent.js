import { useIntl } from 'react-intl'
import { useCallback } from 'react'
import Select from '../shared/Select'
import Pill from '../shared/Pill'
import Checkbox from '../shared/Checkbox'

export const WizardStage1 = ({ wizardData, allValues, setAllValues }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const addSector = (sector) =>
    setAllValues(prevValues => ({
      ...prevValues,
      sectors: [...allValues.sectors.filter(({ slug }) => slug !== sector.slug), sector]
    }))

  const removeSector = (sector) =>
    setAllValues(prevValues => ({
      ...prevValues,
      sectors: allValues.sectors.filter(value => value !== sector)
    }))

  const addSdg = (sdg) =>
    setAllValues(prevValues => ({
      ...prevValues,
      sdgs: [...allValues.sdgs.filter(({ slug }) => slug !== sdg.slug), sdg]
    }))

  const removeSdg = (sdg) =>
    setAllValues(prevValues => ({
      ...prevValues,
      sdgs: allValues.sdgs.filter(value => value !== sdg)
    }))

  return (
    <div className='grid xl:grid-cols-3 gap-x-12 gap-y-3'>
      <div className='flex flex-col gap-3 order-1'>
        <div className='text-sm'>
          <div>{format('wizard.selectSector')}</div>
        </div>
        <Select
          className='mt-auto'
          options={wizardData.sectors}
          value={null}
          onChange={(sector) => addSector(sector)}
          placeholder={format('wizard.sectorPlaceholder')}
        />
      </div>
      <div className='flex flex-col gap-3 order-3 xl:order-2'>
        <div className='text-sm'>
          {format('wizard.selectUseCase')}
        </div>
        <Select
          className='mt-auto'
          options={wizardData.useCases}
          value={
            allValues.useCase &&
            { value: allValues.useCase, label: allValues.useCase }
          }
          onChange={(useCase) =>
            setAllValues(prevValues =>
              ({ ...prevValues, useCase: useCase?.label ?? '' }))
          }
          placeholder={format('wizard.useCasePlaceholder')}
          isClearable
        />
      </div>
      <div className='flex flex-col gap-3 order-5 xl:order-3'>
        <div className='text-sm'>
          {format('wizard.selectSDG')}
        </div>
        <Select
          options={wizardData.sdgs}
          value={null}
          onChange={(sdg) => addSdg(sdg)}
          placeholder={format('wizard.sdgPlaceholder')}
        />
      </div>
      <div className='pill-container order-2 xl:order-4'>
        <div className='flex flex-wrap gap-3'>
          {allValues.sectors?.map((sector, sectorIdx) => (
            <Pill
              key={`sector-${sectorIdx}`}
              label={sector.label}
              onRemove={() => removeSector(sector)}
            />
          ))}
        </div>
      </div>
      <div className='order-4 xl:order-5'/>
      <div className='pill-container order-6'>
        <div className='flex flex-wrap gap-3'>
          {allValues.sdgs?.map((sdg, sdgIdx) => (
            <Pill
              key={`sdg-${sdgIdx}`}
              label={sdg.label}
              onRemove={() => removeSdg(sdg)}
            />
          ))}
        </div>
      </div>
    </div>
  )
}

export const WizardStage2 = ({ wizardData, allValues, setAllValues }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const addSelectedTagValue = (selectedTagValue) => {
    if (allValues.tags.indexOf(selectedTagValue) < 0) {
      allValues.tags.push(selectedTagValue)
      setAllValues(prevValues => ({
        ...prevValues,
        tags: allValues.tags
      }))
    }
  }

  const addSelectedCountryName = (selectedCountryName) => {
    if (allValues.countries.indexOf(selectedCountryName) < 0) {
      allValues.countries.push(selectedCountryName)
      setAllValues(prevValues => ({
        ...prevValues,
        countries: allValues.countries
      }))
    }
  }

  const removeTag = (tagValue) => {
    setAllValues(prevValues => ({
      ...prevValues,
      tags: allValues.tags.filter(val => val !== tagValue)
    }))
  }

  const removeCountry = (countryValue) => {
    setAllValues(prevValues => ({
      ...prevValues,
      countries: allValues.countries.filter(val => val !== countryValue)
    }))
  }

  return (
    <div className='flex flex-col lg:flex-row gap-x-12 gap-y-3'>
      <div className='lg:w-1/3'>
        <div className='text-sm my-2'>
          {format('wizard.selectTags')}
        </div>
        <Select
          options={wizardData.tags}
          onChange={(tag) => addSelectedTagValue(tag.value)}
          placeholder={format('wizard.tagPlaceholder')}
          value={null}
        />
        <div className='flex flex-wrap gap-3 mt-4'>
          {allValues.tags.map((tag, tagIdx) => (
            <Pill
              key={`tag-${tagIdx}`}
              label={tag}
              onRemove={() => removeTag(tag)}
            />
          ))}
        </div>
      </div>
      <div className='lg:w-1/3'>
        <div className='text-sm my-2'>
          {format('wizard.selectCountry')}
        </div>
        <Select
          options={wizardData.countries}
          onChange={(country) => addSelectedCountryName(country.value)}
          placeholder={format('wizard.countryPlaceholder')}
          value={null}
        />
        <div className='flex flex-wrap gap-3 mt-4'>
          {allValues.countries.map((country, countryIdx) => (
            <Pill
              key={`country-${countryIdx}`}
              label={country}
              onRemove={() => removeCountry(country)}
            />
          ))}
        </div>
      </div>
      <div className='lg:w-1/3'>
        <div className='text-sm my-2'>
          {format('wizard.selectMobile')}
        </div>
        <div className='grid xl:grid-cols-2 gap-2'>
          {wizardData.mobileServices.map((service, serviceIdx) => (
            <div key={serviceIdx} className='text-sm'>
              <label>
                <Checkbox
                  onChange={(e) => e.currentTarget.checked
                    ? allValues.mobileServices.push(service.value) &&
                        setAllValues(prevValues => ({
                          ...prevValues,
                          mobileServices: allValues.mobileServices
                        }))
                    : setAllValues(prevValues => ({
                      ...prevValues,
                      mobileServices: allValues.mobileServices.filter(val => val !== service.value)
                    }))
                  }
                />
                <span className='pl-2'>
                  {service.label}
                </span>
              </label>
            </div>
          ))}
        </div>
      </div>
    </div>
  )
}

export const WizardStage3 = ({ wizardData, allValues, setAllValues }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])
  const classNameSelected = 'border border-dial-stratos bg-dial-stratos text-white rounded p-4'
  const classNameNotSelected = 'border border-dial-stratos text-dial-stratos rounded p-4 '

  return (
    <div className='flex flex-col lg:flex-row gap-x-12 gap-y-3'>
      <div className='lg:w-1/3'>
        <div className='text-sm'>
          {format('wizard.buildingBlocks')}
        </div>
      </div>
      <div className='lg:w-2/3'>
        <div className='text-sm bb-content overflow-y-auto flex flex-col gap-4'>
          {wizardData.buildingBlocks.map((bb) => {
            return (
              <div key={bb.id} className='flex flex-row gap-4 items-center'>
                <button
                  onClick={() => {
                    allValues.buildingBlocks.push(bb.name) &&
                    setAllValues(prevValues => ({
                      ...prevValues,
                      buildingBlocks: allValues.buildingBlocks
                    }))
                  }}
                  className={allValues.buildingBlocks.includes(bb.name)
                    ? classNameSelected
                    : classNameNotSelected
                  }
                >
                  {format('wizard.yes')}
                </button>
                <button
                  onClick={() => setAllValues(
                    prevValues => ({
                      ...prevValues,
                      buildingBlocks: allValues.buildingBlocks.filter(val => val !== bb.name)
                    })
                  )}
                  className={allValues.buildingBlocks.includes(bb.name)
                    ? classNameNotSelected
                    : classNameSelected
                  }
                >
                  {format('wizard.no')}
                </button>
                <div className='inline-block my-2'>
                  {bb.name.toUpperCase()}
                  <span className='ml-1 capitalize'>({bb.maturity.toLowerCase()})</span>
                  <br />
                  {format('wizard.bb.' + bb.name.replace(/\s+/g, '').toLowerCase())}
                </div>
              </div>
            )
          })}
        </div>
      </div>
    </div>
  )
}
