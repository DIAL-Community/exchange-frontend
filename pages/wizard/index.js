import { NextSeo } from 'next-seo'
import { useCallback } from 'react'
import { useIntl } from 'react-intl'
import ClientOnly from '../../lib/ClientOnly'
import Footer from '../../components/shared/Footer'
import Header from '../../components/shared/Header'
import WizardMain from '../../components/wizard/WizardMain'
import { WizardContextProvider } from '../../components/wizard/WizardContext'

const Wizard = ({ defaultTenants }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  return (
    <>
      <NextSeo
        title={format('app.title')}
        description={format('seo.description.about')}
      />
      <Header />
      <ClientOnly clientTenants={defaultTenants}>
        <WizardContextProvider>
          <WizardMain />
        </WizardContextProvider>
      </ClientOnly>
      <Footer />
    </>
  )
}

export async function getServerSideProps() {
  const response = await fetch(process.env.NEXTAUTH_URL + '/api/tenants')
  const { defaultTenants } = await response.json()

  // Passing data to the page as props
  return { props: { defaultTenants } }
}

export default Wizard
