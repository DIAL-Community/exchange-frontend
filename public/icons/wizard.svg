<svg xmlns="http://www.w3.org/2000/svg" width="30.849" height="30.854" viewBox="0 0 30.849 30.854">
    <defs>
        <style>
            .cls-1{fill:#323245;fill-rule:evenodd}
        </style>
    </defs>
    <g id="Group_864" transform="translate(-41.723 -.722)">
        <path id="Path_489" d="M38.233 35.508l.011-.011 11.741 11.1-4.433 4.69-11.693-11.055-.011.011-8.312-7.857 4.431-4.69zm-9.413-3.215l4.936 4.664L35 35.646l-4.94-4.665q-.618.657-1.239 1.312z" class="cls-1" transform="translate(22.587 -19.708)"/>
        <path id="Path_490" d="M57.086 12.688l-4.754 5.029 2.028 1.917 4.754-5.034z" class="cls-1" transform="translate(6.171 -10.513)"/>
        <path id="Path_491" d="M18.958 53.768L14.2 58.8l2.028 1.917 4.754-5.029z" class="cls-1" transform="translate(29.53 -35.681)"/>
        <path id="Path_492" d="M16.954 16.25l-5.029-4.75-1.917 2.027 5.029 4.754z" class="cls-1" transform="translate(32.1 -9.783)"/>
        <path id="Path_493" d="M13.775 40.1l-4.763.4.167 1.979 4.761-.4z" class="cls-1" transform="translate(32.711 -27.307)"/>
        <path id="Path_494" d="M38.772 13.7l.328-4.764 1.982.136-.327 4.767z" class="cls-1" transform="translate(14.478 -8.214)"/>
        <path id="Path_495" d="M39.9 63.616l.011 4.78h-1.986l-.012-4.78z" class="cls-1" transform="translate(15.005 -41.714)"/>
        <path id="Path_496" d="M62.607 38.386l4.777-.141-.059-1.985-4.777.139z" class="cls-1" transform="translate(-.088 -24.955)"/>
    </g>
</svg>
