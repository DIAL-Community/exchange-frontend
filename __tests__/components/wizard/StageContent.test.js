import { mockRouterImplementation, render } from '../../test-utils'
import { WizardStage1, WizardStage2 } from '../../../components/wizard/StageContent'
import CustomMockedProvider from '../../utils/CustomMockedProvider'
import { projData, allValues } from './data/WizardStage'

jest.mock('next/dist/client/router')

describe('Unit test for the WizardStage1 component.', () => {

  beforeAll(() => {
    mockRouterImplementation()
  })

  test('Should match snapshot.', () => {
    const { container } = render(
      <CustomMockedProvider>
        <WizardStage1
          projData={projData}
          allValues={allValues}
        />
      </CustomMockedProvider>
    )
    expect(container).toMatchSnapshot()
  })
})

describe('Unit test for the WizardStage2 component.', () => {

  beforeAll(() => {
    mockRouterImplementation()
  })

  test('Should match snapshot.', () => {
    const { container } = render(
      <CustomMockedProvider>
        <WizardStage2
          projData={projData}
          allValues={allValues}
        />
      </CustomMockedProvider>
    )
    expect(container).toMatchSnapshot()
  })
})
