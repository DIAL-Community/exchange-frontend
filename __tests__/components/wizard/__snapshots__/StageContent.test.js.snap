// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for the WizardStage1 component. Should match snapshot. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="lg:flex gap-12"
  >
    <div
      class="lg:w-1/3"
    >
      <div
        class="text-sm my-2"
      >
        <div>
          Select the sector that your project supports
        </div>
      </div>
      <div
        class=" css-l45kjq-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-:r0:-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class="react-select__control css-qv4grc-control"
        >
          <div
            class="react-select__value-container css-17t7gkf-ValueContainer"
          >
            <div
              class="react-select__placeholder css-5rjclb-placeholder"
              id="react-select-:r0:-placeholder"
            >
              Select a Sector
            </div>
            <div
              class="react-select__input-container css-qbdosj-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-:r0:-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class="react-select__input"
                id="react-select-id-undefined"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-wrap gap-3 mt-4"
      >
        <div
          class="shadow-md flex gap-2 px-2 py-1 bg-white text-dial-stratos"
          data-testid="pill"
        >
          sector 1
          <svg
            class="cursor-pointer opacity-50"
            data-testid="remove-button"
            fill="currentColor"
            height="1.5em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1.5em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </div>
        <div
          class="shadow-md flex gap-2 px-2 py-1 bg-white text-dial-stratos"
          data-testid="pill"
        >
          sector 2
          <svg
            class="cursor-pointer opacity-50"
            data-testid="remove-button"
            fill="currentColor"
            height="1.5em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1.5em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="lg:w-1/3"
    >
      <div
        class="text-sm my-2"
      >
        Select a specific use case, if applicable
      </div>
      <div
        class=" css-l45kjq-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-:r1:-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class="react-select__control css-qv4grc-control"
        >
          <div
            class="react-select__value-container react-select__value-container--has-value css-17t7gkf-ValueContainer"
          >
            <div
              class="react-select__single-value css-1dimb5e-singleValue"
            >
              useCase
            </div>
            <div
              class="react-select__input-container css-qbdosj-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class="react-select__input"
                id="react-select-id-undefined"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              class="react-select__indicator react-select__clear-indicator css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                />
              </svg>
            </div>
            <span
              class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="lg:w-1/3"
    >
      <div
        class="text-sm my-2"
      >
        Select any SDG that the project is designed to support
      </div>
      <div
        class=" css-l45kjq-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-:r2:-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class="react-select__control css-qv4grc-control"
        >
          <div
            class="react-select__value-container css-17t7gkf-ValueContainer"
          >
            <div
              class="react-select__placeholder css-5rjclb-placeholder"
              id="react-select-:r2:-placeholder"
            >
              Select an SDG
            </div>
            <div
              class="react-select__input-container css-qbdosj-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-:r2:-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class="react-select__input"
                id="react-select-id-undefined"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-wrap gap-3 mt-4"
      >
        <div
          class="shadow-md flex gap-2 px-2 py-1 bg-white text-dial-stratos"
          data-testid="pill"
        >
          sdg 1
          <svg
            class="cursor-pointer opacity-50"
            data-testid="remove-button"
            fill="currentColor"
            height="1.5em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1.5em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </div>
        <div
          class="shadow-md flex gap-2 px-2 py-1 bg-white text-dial-stratos"
          data-testid="pill"
        >
          sdg 2
          <svg
            class="cursor-pointer opacity-50"
            data-testid="remove-button"
            fill="currentColor"
            height="1.5em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1.5em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the WizardStage2 component. Should match snapshot. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="lg:flex gap-12"
  >
    <div
      class="lg:w-1/3"
    >
      <div
        class="text-sm my-2"
      >
        Select any tags that are applicable to this project
      </div>
      <div
        class=" css-l45kjq-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-:r3:-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class="react-select__control css-qv4grc-control"
        >
          <div
            class="react-select__value-container css-17t7gkf-ValueContainer"
          >
            <div
              class="react-select__placeholder css-5rjclb-placeholder"
              id="react-select-:r3:-placeholder"
            >
              Select a Tag
            </div>
            <div
              class="react-select__input-container css-qbdosj-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-:r3:-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class="react-select__input"
                id="react-select-id-undefined"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-wrap gap-3 mt-4"
      />
    </div>
    <div
      class="lg:w-1/3"
    >
      <div
        class="text-sm my-2"
      >
        What country(s) will this project be implemented in?
      </div>
      <div
        class=" css-l45kjq-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-:r4:-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class="react-select__control css-qv4grc-control"
        >
          <div
            class="react-select__value-container css-17t7gkf-ValueContainer"
          >
            <div
              class="react-select__placeholder css-5rjclb-placeholder"
              id="react-select-:r4:-placeholder"
            >
              Select a Country
            </div>
            <div
              class="react-select__input-container css-qbdosj-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-:r4:-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class="react-select__input"
                id="react-select-id-undefined"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-wrap gap-3 mt-4"
      />
    </div>
    <div
      class="lg:w-1/3"
    >
      <div
        class="text-sm my-2"
      >
        Will mobile services be required for this project?
      </div>
      <div
        class="xl:grid xl:grid-cols-2 gap-2"
      >
        <div
          class="text-sm"
        >
          <label>
            <input
              class="h-8 w-8"
              data-testid="checkbox"
              type="checkbox"
            />
            <span
              class="pl-2"
            />
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
